build.override.targets

```diff
@@ -0,0 +1,23 @@
+<?xml version="1.0" encoding="utf-8"?>
+<Project ToolsVersion="4.0"  xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+
+  <PropertyGroup>
+    <TraversalBuildDependsOn>
+      WcfSetup;
+      $(TraversalBuildDependsOn);
+      WcfCleanup;
+    </TraversalBuildDependsOn>
+  </PropertyGroup>
+  <!--if ServiceUri is set, the service has been already started on another machine-->
+  <Target Name="WcfSetup" Condition="($(WithCategories.Contains('OuterLoop')) or '$(OuterLoop)' == 'true') and ('$(ServiceUri)' == '') and $(OSEnvironment) == 'Windows_NT'" >
+    <Exec Command="$(MSBuildThisFileDirectory)\src\System.Private.ServiceModel\tools\scripts\StartWCFSelfHostedSvc.cmd">
```

Comment:
If we detect the exit code is other than the ones we expect (0 or -1), then I believe we can use the Error task to log a message and fail the build.  https://msdn.microsoft.com/en-us/library/8b08t3s4.aspx .